src/Components/App/App.js

```diff
@@ -1,19 +1,43 @@
 import "./App.css"
+import React, { useEffect, useState } from "react"
 import { Route, Routes, NavLink } from "react-router-dom"
 import { useSelector, useDispatch } from "react-redux"
-import { increment, decrement } from "../../actions"
-import Header from "../../Components/Header/Header"
+import { assignUsers, assignTrails } from "../../actions"
+import Header from "../Header/Header"
 import Welcome from "../Welcome/Welcome"
 import AboutUs from "../AboutUs/AboutUs"
 import SavedTrails from "../SavedTrails/SavedTrails"
 import IndividualTrail from "../IndividualTrail/IndividualTrail"
 import AllTrails from "../AllTrails/AllTrails";
+import Login from "../Login/Login"
 import { Form } from "react-router-dom";
 
-function App() {
-  const counter = useSelector((state) => state.counter)
+const App = () => {
+  const users = useSelector((state) => state.users)
   const dispatch = useDispatch()
 
+  const getData = (url) => {
+    return fetch(url, {
+    })
+      .then(response => response.json())
+      .catch(err => console.log(err))
+  }
+
+  const instantiateData = () => {
+    Promise.all([
+      getData('http://localhost:3000/api/v1/users'),
+      getData('http://localhost:3000/api/v1/trails')
```

Comment:
Here is a great spot to have an environment variable.

When developing locally, running the API calls against `localhost` makes a lot of sense. But once the app is deployed to an environment, be it staging or production, you don't want hardcoded `localhost` calls in your code.

Using an environment variable allows you to inject the proper URL at runtime, so you only need to code the call once.